test/integration/glue_test.py

```diff
@@ -1,121 +1,229 @@
+    pudl_out: PudlTabl,
+    save_unmapped_ids: bool,
+    test_dir,
+    request,
+):
+    """Test that the stored ids are internally consistent. Label and save (optionally).
+
+    Args:
+        ids_left: name of fixure cooresponding to a dataframe which contains ID's
+        ids_right: name of fixure cooresponding to a dataframe which contains ID's
+        id_cols: list of ID column(s)
+        label_func: If a labeling function is provided, label the missing ID's with
+            flags and columns needed for manual mapping
+        pudl_out: an instance of a pudl output object
+        save_unmapped_ids: If ``True``, export any missing ID's.
+        test_dir: path to the ``test`` directory. Will be used to construct path to the
+            ``devtools/ferc1-eia-glue`` directory to save outputs into.
+
+    Raises:
+        AssertionError:
+    """
+    missing = get_missing_ids(
+        glue_test_dfs[ids_left],
+        glue_test_dfs[ids_right],
+        id_cols,
     )
-    # Use that index to select only the previously unmapped utilities:
-    unmapped_utilities = utils_with_unmapped_plants.loc[
-        new_utilities_index
-    ].reset_index()
-    if len(unmapped_utilities) != 1:
-        raise AssertionError(
-            f"Found {len(unmapped_utilities)} "
-            f"unmapped FERC 1 utilities instead of 1."
-        )
+    if label_func:
+        missing = label_func(missing, pudl_out)
+    if save_unmapped_ids:
+        save_to_devtools_glue(df=missing, test_dir=test_dir, request=request)
+    if not missing.empty:
+        raise AssertionError(f"Found {len(missing)} {id_cols}: {missing}")
 
 
-def test_unmapped_plants_eia(pudl_settings_fixture, pudl_engine):
-    """Check for unmapped EIA Plants."""
-    unmapped_plants_eia = get_unmapped_plants_eia(pudl_engine)
-    if not unmapped_plants_eia.empty:
-        raise AssertionError(
-            f"Found {len(unmapped_plants_eia)} unmapped EIA plants, expected 0."
-            f"{unmapped_plants_eia}"
-        )
+@pytest.mark.parametrize(
+    "ids_left,ids_right,id_cols,drop",
+    [
+        pytest.param(
+            "plants_ferc1",
+            "plants_ferc1_raw",
+            ["utility_id_ferc1", "plant_name_ferc1"],
+            (227, "comanche"),
+            id="check_for_unmmapped_plants_in_plants_ferc1",
+        ),
+        pytest.param(
+            "utilities_ferc1",
+            "utilities_ferc1_xbrl",
+            ["utility_id_ferc1"],
+            (227),
+            id="validate_utility_id_ferc1_in_utilities_ferc1_xbrl",
+        ),
+    ],
+)
+def test_for_unmapped_ids_minus_one(
+    ids_left: str,
+    ids_right: str,
+    id_cols: list[str],
+    drop: tuple,
+    glue_test_dfs: dict[str, pd.DataFrame],
+):
+    """Test that we will find one unmapped ID after dropping one.
+
+    Args:
+        ids_left: name of fixure cooresponding to a dataframe which contains ID's
+        ids_right: name of fixure cooresponding to a dataframe which contains ID's
```

Comment:
I think these need to be updated to reflect the move away from using individual fixtures.

Also cooresponding -> corresponding